### PR TITLE
Significant refactoring with updated GUI.

### PR DESCRIPTION
**Pull request by suderman-google**
Continued refactor.

### CODE REVIEW

File: spatialmedia/__init__.py [hunk 1]
```diff
@@ -0,0 +1,26 @@
...
+
+import os
+import sys
+
+__all__ = ["mpeg", "spherical"]
+
+import spatialmedia.mpeg
+import spatialmedia.spherical
+
+sys.path.append(os.path.dirname(os.path.abspath(__file__)))
```
Comment (jprouty) [inline spatialmedia/__init__.py]: Add a comment; e.g. Ensure this package is installed & available as a top-level package.

File: spatialmedia/gui.py [hunk 2]
```diff
@@ -0,0 +1,118 @@
...
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+"""Spherical Metadata Python Tool
+
+Tool for examining and injecting spherical metadata into MP4 files.
+Naming conventions are styled to match TKinter naming conventions.
```
Comment (jprouty) [inline spatialmedia/gui.py]: I think you can drop this comment as your naming below is both PEP-8 and Google style complaint. You renamed createWidgets -> create_widgets, making it all happy.

File: spatialmedia/mpeg/box.py [hunk 3]
```diff
@@ -0,0 +1,201 @@
...
+      box: box, stco/co64 box to copy.
+      mode: string, bit packing mode for index entries.
+      mode_length: int, number of bytes for index entires.
+      delta: int, offset change for index entries.
+    """
+    fh = in_fh
+    if not box.contents:
+        fh.seek(box.content_start())
+    else:
+        fh = StringIO.StringIO(box.contents)
```
Comment (jprouty) [inline spatialmedia/mpeg/box.py]: This module does not import StringIO. Please add.

File: spatialmedia/mpeg/mpeg4.py [hunk 4]
```diff
@@ -0,0 +1,132 @@
...
+        loaded_mpeg4.first_mdat_box.header_size
+
+    loaded_mpeg4.content_size = 0
+    for element in loaded_mpeg4.contents:
+        loaded_mpeg4.content_size += element.size()
+
+    return loaded_mpeg4
+
+
+class Mpeg4(container.Container):
```
Comment (jprouty) [inline spatialmedia/mpeg/mpeg4.py]: Mpeg4 seems so vague to me. Especially being mpeg4/mpeg4.py. It's not very indicative of it's use. I think Mpeg4Container (and mpeg4_container.py) adds a bit more context and more closely matches the doc string.

File: spatialmedia/spherical.py [hunk 5]
```diff
@@ -0,0 +1,377 @@
...
+                if added:
+                    if not element.add(spherical_uuid(metadata)):
+                        return False
+                    break
+
+    mpeg4_file.resize()
+    return True
+
+
+def ParseSphericalXML(contents, console):
```
Comment (jprouty) [inline spatialmedia/spherical.py]: function naming should be consistent within this file and package: use small_case. If you're trying to distinguish between public/private, a leading underscore is allowable in pep-8/google styles.

File: spatialmedia/spherical.py [hunk 6]
```diff
@@ -0,0 +1,377 @@
...
+                            contents = sub_element.contents[16:]
+                        else:
+                            contents = fh.read(sub_element.content_size - 16)
+                        metadataSets[trackName] = \
+                            ParseSphericalXML(contents, console)
+    return metadataSets
+
+
+def ParseMpeg4(input_file, console):
+    in_fh = open(input_file, "rb")
```
Comment (jprouty) [inline spatialmedia/spherical.py]: When doing file IO such as this, it's more pythonic to use the with syntax:

with open('filename', 'mode') as f:
  ...

See the bottom of this section for example/rationale:
https://docs.python.org/2/tutorial/inputoutput.html#methods-of-file-objects

**Comment by jprouty** [comment]
Thanks Rob! This is looking much much better now. Just a few last comments (already sent I suppose?!) and we're groovy.

File: spatialmedia/gui.py (hunk 2 above)
Comment (suderman-google) [inline spatialmedia/gui.py]: Done.

File: spatialmedia/mpeg/box.py (hunk 3 above)
Comment (suderman-google) [inline spatialmedia/mpeg/box.py]: Done.

**Comment by suderman-google** [comment]
All done.

**Pull request merged**